guides-markdown/multisig-wallet.md

```diff
@@ -0,0 +1,370 @@
+> If the wallet does, then it is able to derive account private key so the account will be *hot* .
+> In fact, this is the way how bcoin actually handles it's private key for an account, it derives every time when it's need.
+> If the wallet doesn't hold master private key (i.e. if it's watch-only), then an account derived from the wallet will be also watch-only.
+
+we can set watch-only wallet named `sharedWallet1` on bcoin node by
+
+```bash
+curl -X PUT $url/wallet/sharedWallet1 --data '
+  {"watchOnly": true, "type": "multisig", "m": 2, "n": 3 }
+  '
+
+# or by cli tool (installed with `npm i -g bclient`)
+bwallet-cli \
+  --url=${url} \
+  --network=testnet \
+  --api-key=mySecretApiKey \
+  --token=${myadmintoken} \ # Must use your admin token to create a new wallet
+  --watch-only=true \ # make it watch only
+  --type=multisig \
+  --m=2 \
+  --n=3 \
+  mkwallet sharedWallet1
+```
+
+Wallet will always have account named `default` when it's created.
+This is the multisig account we are going to use.
+
+> NOTE: this time, we specified `type` for the wallet. But strictly speaking again, wallet itself does not hold property if it is multisig or not.
+> So some options we have passed here (i.e. `type`, `m`, and `n`) are for the default account created with the wallet.
+
+If the wallet is created successfully, than you will see the access token for that wallet in the return value.
+So let's set that token into variable named `sharedWalletToken` and pass to co-signers.
+
+You must register your public key to this `default` account. Let's do it by using bcoin api.
+
+TODO: create and register key.
+
+```js
+
+```
+
+Next, cosigner Alice should create tpub too and set it to the account.
+Although it is possible that Alice can create it in the same way you did,
+lets assume that Alice uses [bitcoinjs-lib](https://github.com/bitcoinjs/bitcoinjs-lib) in her software.
+So the code will be something like ...
+
+```js
+const bjs = require("bitcoinjs-lib")
+const masterHD = new bjs.bip32.fromSeed(Buffer.from("ffffffffffffffffffffffffffffffff", "hex"), bjs.networks.testnet)
+// We are using segwit so following bip84
+const tpub = masterHD.derivePath("m/84'/0'/3'").neutered().toBase58()
+console.log(tpub)
+```
+
+and throw it to the bcoin server
+
+```bash
+curl -X PUT ${url}/wallet/sharedWallet1/shared-key\?token\=${sharedWalletToken} \
+  --data \ 
+  '{
+    "account": "default",
+    "accountKey": '"${aliceTpub}"',
+    }'
+```
+
+> NOTE: It is possible to delete the xpub(tpub) from the wallet if the number of keys registered hasn't reached to `n`.
+> This could be useful when you accidentally registered a wrong key.
+> But if it has reached to `n` , you must create a new account.
+
+Finally, do the same thing for Bob. Then the multisig account is ready.
+
+### 4. Receive incoming information from the wallet server.
+
+Ok, so let's check the wallet's status by
+
+```bash
```

Comment:
Are you familiar with `jq`? It lets you parse json from stdin and could be very useful for giving precise commands and their results

```
$ curl ${url}/wallet/sharedWallet1/account/default?token=${sharedWalletToken} \
    | jq .balance

{
    "tx": 0,
    "coin": 0,
    "unconfirmed": 0,
    "confirmed": 0
}
```